core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -213,6 +215,12 @@ public RelDataType toType(RelDataTypeFactory typeFactory, Object o) {
             Util.enumVal(SqlTypeName.class, (String) map.get("type"));
         final Integer precision = (Integer) map.get("precision");
         final Integer scale = (Integer) map.get("scale");
+        if (SqlTypeName.INTERVAL_TYPES.contains(sqlTypeName)) {
+          TimeUnit startUnit = sqlTypeName.getEndUnit();
```

Comment:
Shouldn't it be getStartUnit()?

Reply:
yes, it should.
Thanks for pointing it out.